CMakeLists.txt

```diff
@@ -948,6 +948,10 @@ if(XCODE)
   swift_common_xcode_cxx_config()
 endif()
 
+# Check what linux distribution is being used.
+# This can be used to determine the default linker to use.
+cmake_host_system_information(RESULT DISTRO_NAME  QUERY DISTRIB_PRETTY_NAME)
```

Comment:
This should be reverted.

The minimum CMake version is 3.19.6 (says so at the top of the file).

This key only appeared on CMake 3.22.

Reply:
The minimum cmake version required for the swift toolchain build has been `"cmake": "v3.24.2"` since version 5.10

https://github.com/apple/swift/blob/main/utils/update_checkout/update-checkout-config.json#L232

https://github.com/apple/swift/pull/67018

Reply:
That's the version used by the `build-script` for CI, and only on Linux and BSD. The CMake files should be compatible with the version required by the `cmake_minimum_required` instruction. This is failing for a macOS which a 3.20 version of CMake (enough to build LLVM and Swift using only CMake).

Reply:
I think the best option here is to update the minimum cmake version to match the version used in the CI. 
This was supposed to have been done already and it makes no sense to be using a cmake version as old as v3.19.6

Reply:
I think @etcwilde plans to significantly increase the minimum CMake version this year anyway, so it would make sense to increase it to 3.22 now itself.

Evan, wdyt?

Reply:
But things need to be done in certain order, and not as a reaction to introducing a dependency to check for a specific Linux distribution, and have a broken `CMakeLists.txt` at the root of the project.

Reply:
Assuming #73001 fixes the problem for now, we can discuss the CMake version separately.